### ISSUE TITLE
WSGI flask api not accessible from apache ssl

### ISSUE THREAD

**Issue by girum89** [GrahamDumpleton/mod_wsgi#668]
I had a working flask api with http but when I moved to https I am not geting a response. the previous working conf is flask.conf
[flask.txt](https://github.com/GrahamDumpleton/mod_wsgi/files/6262239/flask.txt)

and now I have 3 files in my apache sites enabled with the modified flask-ssl.conf file(after a2ensite and restart). I could not access my api. I just want only https requests. Here are my files. Thank you.

[000-default.txt](https://github.com/GrahamDumpleton/mod_wsgi/files/6262302/000-default.txt)
[ds-ssl.txt](https://github.com/GrahamDumpleton/mod_wsgi/files/6262343/ds-ssl.txt)
[flask-ssl.txt](https://github.com/GrahamDumpleton/mod_wsgi/files/6262344/flask-ssl.txt)

Note: I `replaced my domain with aa.example.com` for this question. I can access the home page from my domain but my api links are not working. Thank you

**Comment by GrahamDumpleton** [comment]
Define "api links are not working".

What error are you seeing and where?

What is the HTTP client you are using when making calls against the API?

Are you using a self signed certificate?

Is the HTTP client when calling the API being told to trust an insecure certificate if you are using a self signed certificate?

**Comment by girum89** [comment]
I am using third party certificate. (digicert.com)
I use flask to create an api and I previusly accessed my api at `aa.example.com/gspapi` other links follow the same path. I use Angular HTTP client to make http calls. Now for testing am using `postman`.

**Comment by GrahamDumpleton** [comment]
But as I asked, define "api links are not working". What error are you seeing and where?

It is impossible for me to guess what problem you are actually seeing if you can't explain it better to me.

Have you tried accessing the URL for the API endpoints using curl on the command line rather than a browser based HTTP client?

Have you tried accessing the URL from the same host as Apache is running instead of a separate host?

**Comment by girum89** [comment]
Thank for the prompt responses.

my api links are for backend and frontend communication. I send data from my website to backend by api. 
the only error I am getting from my error.log is  `mod_md support is unavailable.` May be its related to apache and ssl configuration. I will dig more and inform you. Thanks

**Comment by girum89** [comment]
File `000-default.conf` is used for redirection. `ds-ssl.conf `acts as the default ssl config file. my `flask-ssl.conf` file although being enabled it  is not accessed. when I disableed the `ds-ssl.conf`, the `flask-ssl.conf` becomes the default virtual host and WGSI works. I know its an apache issue but I do not understand why they are not working together as `ds-ssl.conf` doen't have ServerName.

**Comment by GrahamDumpleton** [comment]
You aren't then using a hostname in the URL for the request which matches the ServerName/ServerAlias of flask-ssl.conf VirtualHost. When it can't match it will use the default VirtualHost. I already pointed out you had a wrong ServerName in other issue, but then who knows since you aren't including the original configs but ones which have been changed, so I can't verify.

Note that if you are using some custom HTTP client and that client also isn't correctly setting the Host header as well, then the name based virtual hosting of Apache will not work.
